server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DomainMappingsRoutesTest.java

```diff
@@ -0,0 +1,217 @@
+        webAdminServer = WebAdminUtils.createWebAdminServer(new DefaultMetricFactory(), domainMappingsRoutes);
+        webAdminServer.configure(NO_CONFIGURATION);
+        webAdminServer.await();
+
+        RestAssured.requestSpecification = WebAdminUtils.buildRequestSpecification(webAdminServer)
+                .setBasePath("/domainMappings")
+                .log(LogDetail.METHOD)
+                .build();
+    }
+
+    @BeforeEach
+    void setUp() throws Exception {
+        recipientRewriteTable = spy(new MemoryRecipientRewriteTable());
+
+        createServer(new DomainMappingsRoutes(recipientRewriteTable, new JsonTransformer()));
+    }
+
+    @AfterEach
+    void tearDown() {
+        webAdminServer.destroy();
+    }
+
+    @Nested
+    class NormalBehaviour {
+
+        @Test
+        void addDomainMappingShouldRespondWithNoContent() {
+            // @formatter:off
+            given(with().body("to.com"))
+                        .put("from.com")
+                    .then()
+                        .statusCode(HttpStatus.NO_CONTENT_204)
+                        .body(isEmptyString());
+            // @formatter:on
+        }
+
+        @Test
+        void getDomainMappings() throws RecipientRewriteTableException {
+            Domain expectedDomain = Domain.of("abc.com");
+            MappingSource mappingSource = MappingSource.fromDomain(expectedDomain);
+            ImmutableList<String> expectedAliases = ImmutableList.of("to_1.com", "to_2.com", "to_3.com");
+
+            for (String alias : expectedAliases) {
+                recipientRewriteTable.addAliasDomainMapping(mappingSource, Domain.of(alias));
+            }
+
+            // @formatter:off
+            final Map<String, List<String>> map =
+                    when()
+                        .get()
+                    .then()
+                        .contentType(ContentType.JSON)
+                        .statusCode(HttpStatus.OK_200)
+                    .extract()
+                        .body()
+                        .jsonPath()
+                        .getMap(".");
+            // @formatter:on
+
+            assertThat(map)
+                    .containsOnly(entry(expectedDomain.name(), expectedAliases));
+        }
+
+        @Test
+        void getDomainMappingsShouldBeEmpty() {
+            // @formatter:off
+            when()
+                .get()
+            .then()
+                .contentType(ContentType.JSON)
+                .statusCode(HttpStatus.OK_200)
+                .body(is("{}"));
+            // @formatter:on
+        }
+
+        @Test
+        void deleteDomainMappingShouldRespondWithNoContent() {
+            // @formatter:off
+            given(with().body("to.com"))
+                        .delete("from.com")
+                    .then()
+                        .statusCode(HttpStatus.NO_CONTENT_204)
+                        .body(isEmptyString());
+            // @formatter:on
+        }
```

Comment:
Could you add these test cases?

Delete should remove domain mappings:
  - Add a domain mapping
  - Remove it
  - Ensure the collection is empty

get domain mappings should group mappings by source domain
 - The same than `getDomainMappings` but with different source domains.

getDomains should filter out mappings without domain mappings
 - Add for instance an address mapping
 - Ensure that the answer is still `{}`